src/main/java/com/google/jenkins/plugins/computeengine/ComputeEngineWindowsLauncher.java

```diff
@@ -0,0 +1,269 @@
+
+import com.trilead.ssh2.Connection;
+import com.trilead.ssh2.HTTPProxyData;
+import com.trilead.ssh2.SCPClient;
+import com.trilead.ssh2.ServerHostKeyVerifier;
+import com.trilead.ssh2.Session;
+import hudson.ProxyConfiguration;
+import hudson.model.TaskListener;
+import hudson.remoting.Channel;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.net.InetSocketAddress;
+import java.net.Proxy;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import jenkins.model.Jenkins;
+
+public class ComputeEngineWindowsLauncher extends ComputeEngineComputerLauncher {
+    public final boolean useInternalAddress;
+
+    //TODO: make this configurable
+    public static final Integer SSH_PORT = 22;
+    public static final Integer SSH_TIMEOUT = 10000;
+    private static final Logger LOGGER = Logger.getLogger(ComputeEngineLinuxLauncher.class.getName());
+    private static int bootstrapAuthTries = 30;
+    private static int bootstrapAuthSleepMs = 15000;
+
+    public ComputeEngineWindowsLauncher(String cloudName, Operation insertOperation, boolean useInternalAddress) {
+        super(cloudName, insertOperation.getName(), insertOperation.getZone());
+        this.useInternalAddress = useInternalAddress;
+    }
+
+    protected void log(Level level, ComputeEngineComputer computer, TaskListener listener, String message) {
+        try {
+            ComputeEngineCloud cloud = computer.getCloud();
+            cloud.log(LOGGER, level, listener, message);
+        } catch (CloudNotFoundException cnfe) {
+            //TODO: figure out how to log without a handle to the cloud
+        }
+    }
+
+    protected void logException(ComputeEngineComputer computer, TaskListener listener, String message, Throwable exception) {
+        try {
+            ComputeEngineCloud cloud = computer.getCloud();
+            cloud.log(LOGGER, Level.WARNING, listener, message, exception);
+        } catch (CloudNotFoundException cnfe) {
+            //TODO: figure out how to log without a handle to the cloud
+        }
+    }
+
+    protected void logInfo(ComputeEngineComputer computer, TaskListener listener, String message) {
+        log(Level.INFO, computer, listener, message);
+    }
+
+    protected void logWarning(ComputeEngineComputer computer, TaskListener listener, String message) {
+        log(Level.WARNING, computer, listener, message);
+    }
+
+    protected void launch(ComputeEngineComputer computer, TaskListener listener, Instance inst)
+            throws IOException, InterruptedException {
+        ComputeEngineInstance node = computer.getNode();
+        if (node == null) {
+            logWarning(computer, listener, "Could not get node from computer");
+            return;
+        }
+
+        final Connection bootstrapConn;
+        final Connection conn;
+        Connection cleanupConn = null; // java's code path analysis for final
```

Comment:
Multi-line comments should use the conventional /* */ syntax.

Reply:
done